students/RimleeT/lesson08/circle.py

```diff
@@ -0,0 +1,61 @@
+import math
+
+class Circle:
+    def __init__(self, radius):
+        self.radius = radius
+
+    @classmethod
+    def from_diameter(cls, diameter):
+        return cls(diameter / 2)
+
+    @property
+    def diameter(self):
+        return self.radius * 2
+    @diameter.setter
+    def diameter(self, diameter):
+        self.radius = diameter / 2
+
+    @property
+    def area(self):
+        return math.pi * self.radius ** 2
+
+    def __str__(self):
+        return "Circle with radius: {:.6f}".format(self.radius)
+
+    def __repr__(self):
+        return "Circle({})".format(round(self.radius))
+
+    def __add__(self, other):
+        return Circle(self.radius + other.radius)
+
+    def __gt__(self, other):
+        return self.radius > other.radius
+
+    def __lt__(self, other):
+        return self.radius < other.radius
+
+    def __eq__(self, other):
+        return self.radius == other.radius
+
+    def __mul__(self, other):
+        if isinstance(other, int):
+            return Circle(self.radius * other)
+
+    __rmul__ = __mul__
+    __iadd__ = __add__
```

Comment:
careful here -- "iadd" means "in-place add" -- when defined for mutable objects, such as Circle, it should alter the existing instance (self), rather than make a new one, so it needs to be defined differently than __add__. A test for this would be:

```
c = Circle(5)
c2 = c  # a new name for the same circle
c + 4
assert c2.radius == 9  # the original Circle should have been changed 
```